src/app/iteration/iteration.service.ts

```diff
@@ -80,10 +81,35 @@ export class IterationService {
   }
 
   /**
-   * Get locally saved iterations' reference
+   * Update an existing iteration
+   * @param iteration - Updated iteration
+   * @return updated iteration's reference from the list
    */
-  getIterationList() {
-    return this.iterations;
+  updateIteration(iteration: IterationModel): Promise<IterationModel> {
+    return this.http
+      .patch(iteration.links.self, { headers: this.headers })
+      .toPromise()
+      .then (response => {
+        if (response.status != 200) {
+          throw new Error('API error occured');
```

Comment:
I think exposing something for unit test purposes, or adjusting the production code for that is a smell. There should be a very strong reason to really go this way.